src/kinetics/Reaction.cpp

```diff
@@ -891,6 +825,329 @@ void BlowersMaselInterfaceReaction::getParameters(AnyMap& reactionNode) const
+        return;
+    }
+
+    parseReactionEquation(*this, node["equation"], kin);
+    // Non-stoichiometric reaction orders
+    if (node.hasKey("orders")) {
+        for (const auto& order : node["orders"].asMap<double>()) {
+            orders[order.first] = order.second;
+            if (kin.kineticsSpeciesIndex(order.first) == npos) {
+                setValid(false);
+            }
+        }
+    }
+
+    // Flags
+    id = node.getString("id", "");
+    duplicate = node.getBool("duplicate", false);
+    allow_negative_orders = node.getBool("negative-orders", false);
+    allow_nonreactant_orders = node.getBool("nonreactant-orders", false);
+
+    calculateRateCoeffUnits(kin);
+    input = node;
+}
+
+std::pair<std::vector<std::string>, bool> Reaction3::undeclaredThirdBodies(
+        const Kinetics& kin) const
+{
+    std::vector<std::string> undeclared;
+    if (m_third_body) {
+        updateUndeclared(undeclared, m_third_body->efficiencies, kin);
+        bool specified_collision_partner = dynamic_cast<const ThreeBodyReaction3*>(
+            this)->specified_collision_partner;
+        return std::make_pair(undeclared, specified_collision_partner);
+    }
+    return std::make_pair(undeclared, false);
+}
+
+void Reaction3::validate()
+{
+    Reaction::validate();
+    m_rate->validate(equation());
+}
+
+ElementaryReaction3::ElementaryReaction3()
+    : Reaction3()
+{
+    m_rate = std::shared_ptr<ArrheniusRate>(new ArrheniusRate);
+}
+
+ElementaryReaction3::ElementaryReaction3(const Composition& reactants,
+                                         const Composition& products,
+                                         const ArrheniusRate& rate)
+    : Reaction3(reactants, products)
+{
+    m_rate = std::make_shared<ArrheniusRate>(rate);
+}
+
+ElementaryReaction3::ElementaryReaction3(const AnyMap& node, const Kinetics& kin)
+    : ElementaryReaction3()
+{
+    setParameters(node, kin);
+    setRate(std::shared_ptr<ArrheniusRate>(new ArrheniusRate(node, rate_units)));
+}
+
+void ElementaryReaction3::getParameters(AnyMap& reactionNode) const
+{
+    Reaction::getParameters(reactionNode);
+    reactionNode.update(m_rate->parameters(rate_units));
+}
+
+ThreeBodyReaction3::ThreeBodyReaction3()
+    : ElementaryReaction3()
+{
+    m_third_body = std::shared_ptr<ThirdBody>(new ThirdBody);
+}
+
+ThreeBodyReaction3::ThreeBodyReaction3(const Composition& reactants,
+                                       const Composition& products,
+                                       const ArrheniusRate& rate,
+                                       const ThirdBody& tbody)
+    : ElementaryReaction3(reactants, products, rate)
+{
+    m_third_body = std::make_shared<ThirdBody>(tbody);
+}
+
+ThreeBodyReaction3::ThreeBodyReaction3(const AnyMap& node, const Kinetics& kin)
+    : ThreeBodyReaction3()
+{
+    setParameters(node, kin);
+    setRate(std::shared_ptr<ArrheniusRate>(new ArrheniusRate(node, rate_units)));
+}
```

Comment:
I honestly believe that `AnyMap` based constructors are useful, despite them being used within `ReactionFactory`. I believe this boils down to a difference in coding style, where I prefer to 'let objects do their things', rather than explicitly assembling objects from the outside. Beyond, I believe it is a way to document how an object is assembled.

The main reason here is that this implementation keeps things simple in `ReactionFactory`, where no further information is necessary. If explicit setters were used, both `setParameters` and `setRate` would have to be called (which are difficult to combine, as there are different overloading strategies). Overall, I have a strong preference to have everything handled by the object rather than adding multiple lines for each object within `ReactionFactory`.

Reply:
OK. Resolving per comment elsewhere.